### PR TITLE
Granular CRN for instance datasources and resources

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_instance (r):

```
=== RUN   TestAccIBMPIInstanceBasic
--- PASS: TestAccIBMPIInstanceBasic (341.98s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceDeploymentType
--- PASS: TestAccIBMPIInstanceDeploymentType (982.16s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceReplicant
--- PASS: TestAccIBMPIInstanceReplicant (1408.88s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceNetwork
--- PASS: TestAccIBMPIInstanceNetwork (1156.62s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceDeploymentTypeNoStorage
--- PASS: TestAccIBMPIInstanceDeploymentTypeNoStorage (288.31s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceUpdateActiveState
--- PASS: TestAccIBMPIInstanceUpdateActiveState (1257.55s)
PASS
```

```
=== RUN   TestAccIBMPIInstanceUpdateStoppedState
--- PASS: TestAccIBMPIInstanceUpdateStoppedState (1248.31s)
PASS
```

```
=== RUN   TestAccIBMPISAPInstance
--- PASS: TestAccIBMPISAPInstance (1284.73s)
PASS
```

pi_cloud_instance (d):

```
=== RUN   TestAccIBMPICloudInstanceDataSource_basic
--- PASS: TestAccIBMPICloudInstanceDataSource_basic (10.04s)
PASS
```

pi_instance (d):

```
=== RUN   TestAccIBMPIInstanceDataSource_basic
--- PASS: TestAccIBMPIInstanceDataSource_basic (24.94s)
PASS
```

pi_instances (d):

```
=== RUN   TestAccIBMPIInstancesDataSource_basic
--- PASS: TestAccIBMPIInstancesDataSource_basic (29.16s)
PASS
```

pi_capture (r):

```
=== RUN   TestAccIBMPICaptureBasic
--- PASS: TestAccIBMPICaptureBasic (51.88s)
PASS
```

```
=== RUN   TestAccIBMPICaptureWithVolume
--- PASS: TestAccIBMPICaptureWithVolume (1263.53s)
PASS
```

```
=== RUN   TestAccIBMPICaptureCloudStorage
--- PASS: TestAccIBMPICaptureCloudStorage (4162.42s)
PASS
```

```
=== RUN   TestAccIBMPICaptureBoth
--- PASS: TestAccIBMPICaptureBoth (1835.85s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 1]
```diff
@@ -404,6 +409,13 @@ func ResourceIBMPIInstance() *schema.Resource {
 				Description: "Fault information.",
 				Type:        schema.TypeMap,
 			},
+			Arg_UserTags: {
+				Description: "List of user tags.",
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:413]: This should be right next to `Arg_UserData` in the arguments section above

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 2]
```diff
@@ -404,6 +409,13 @@ func ResourceIBMPIInstance() *schema.Resource {
...
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+				Optional:    true,
+				Type:        schema.TypeList,
+			},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:418]: This should be right next to `Arg_UserData` in the arguments section above

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 3]
```diff
@@ -77,6 +77,11 @@ func ResourceIBMPIInstance() *schema.Resource {
 				Required:    true,
 				Type:        schema.TypeString,
 			},
+			Attr_CRN: {
+				Computed:    true,
+				Description: "CRN of instance.",
+				Type:        schema.TypeString,
+			},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:84]: This should be in `// Attributes` section below

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 4]
```diff
@@ -1416,6 +1429,10 @@ func createSAPInstance(d *schema.ResourceData, sapClient *instance.IBMPISAPInsta
 	if deploymentTarget, ok := d.GetOk(Arg_DeploymentTarget); ok {
 		body.DeploymentTarget = expandDeploymentTarget(deploymentTarget.(*schema.Set).List())
 	}
+	if tags, ok := d.GetOk(Arg_UserTags); ok {
+		userTags := flex.ExpandStringList(tags.([]interface{}))
+		body.UserTags = userTags
+	}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:1435]: Could you check for `Len()> 0` before expanding it?

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 5]
```diff
@@ -1610,6 +1627,10 @@ func createPVMInstance(d *schema.ResourceData, client *instance.IBMPIInstanceCli
...
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_instance.go:1633]: Same as above comment

**Review by michaelkad — CHANGES_REQUESTED** [review]
Few comments.

**Review by ismirlia — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo

**Pull request closed without merging**